utils/disk_space_monitor.hh

```diff
@@ -0,0 +1,137 @@
+/*
+ * Copyright (C) 2024-present ScyllaDB
+ */
+
+/*
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+#pragma once
+
+#include <chrono>
+#include <filesystem>
+
+#include <boost/signals2/connection.hpp>
+#include <boost/signals2/signal_type.hpp>
+#include <boost/signals2/dummy_mutex.hpp>
+
+#include <seastar/core/abort_source.hh>
+#include <seastar/core/future.hh>
+#include <seastar/core/lowres_clock.hh>
+#include <seastar/core/reactor.hh>
+#include <seastar/core/sleep.hh>
+#include <seastar/core/timer.hh>
+#include <seastar/util/optimized_optional.hh>
+
+#include "seastarx.hh"
+#include "utils/assert.hh"
+#include "utils/log.hh"
+#include "utils/updateable_value.hh"
+
+using namespace std::chrono_literals;
+namespace bs2 = boost::signals2;
+
+namespace utils {
+
+extern seastar::logger dsmlog;
+
+// Instantiated only on shard 0
+template <typename Clock = lowres_clock>
```

Comment:
Why a template? I see you use manual_clock in the test, but I don't understand what you're testing.

Reply:
The test is the only reason for the template indeed.

Reply:
Template dropped in v10 ([9a3db9a](https://github.com/scylladb/scylladb/commit/9a3db9ac9e444e6ef85f52f0223d11cf6d89b4e2))